pkg/internal/layers/layers.go

```diff
@@ -18,36 +21,24 @@ import (
 var MaxConditions = 10
 var rootPath = "/repos"
 
-// Layer is the Schema for the addons API.
-type Layer struct {
-	updated     bool
-	requeue     bool
-	delayed     bool
-	delay       time.Duration
-	ctx         context.Context
-	client      *kubernetes.Clientset
-	log         logr.Logger
-	LayerI      `json:"-"`
-	addonsLayer *kraanv1alpha1.AddonsLayer
-}
-
-// LayerI defines the interface for managing the layer.
-type LayerI interface {
+// Layer defines the interface for managing the layer.
+type Layer interface {
 	StatusReady()
 	StatusApplying()
 	StatusApply()
 	StatusApplyPending()
 	StatusPrunePending()
 	StatusPrune()
 	StatusPruning()
-	StatusDeployed()
+	StatusDeployed(reason, message string)
 	IsHold() bool
 	SetHold()
 	GetStatus() string
 	GetName() string
 	GetNamespace() string
 	GetLogger() logr.Logger
 	GetK8sClient() *kubernetes.Clientset
```

Comment:
Are functions like this needed on the interface? To me it feels like they shouldn't be there.